src/anyio/_backends/_trio.py

```diff
@@ -183,16 +183,17 @@ async def __aexit__(
         exc_val: BaseException | None,
         exc_tb: TracebackType | None,
     ) -> bool | None:
+        rest = None
         try:
             return await self._nursery_manager.__aexit__(exc_type, exc_val, exc_tb)
         except BaseExceptionGroup as exc:
-            _, rest = exc.split(trio.Cancelled)
+            rest = exc.split(trio.Cancelled)[1]
             if not rest:
```

Comment:
Since `rest` isn't referenced outside of the `if` statement, we could just do this:
```suggestion
            # If the exception group only contains Cancelled exceptions, raise a single one to
            # cover them all
            if not exc.split(trio.Cancelled)[1]:
```